### ISSUE TITLE
[Leaving IBM]: leaving ibm component is opening once

### ISSUE THREAD

**Issue by ahmedfarid-dev** [carbon-design-system/carbon-for-ibm-dotcom#7738]
### Description

I am using Leaving IBM component ,it's working fine it opens ,but if I close it from the close icon ,it never opens again 

here's my code 
`{leavingIBMOpen && (
        <Provider store={store}>
          <DDSLeavingIbmContainer href={leavingIBMLink} open />
        </Provider>
      )}`

leavingIBMOpen  is a state flag that I set to true with some button click 

I am using version : "@carbon/ibmdotcom-web-components": "^1.13.0-canary.1445076726.0",

### Component(s) impacted

{leavingIBMOpen && (
        <Provider store={store}>
          <DDSLeavingIbmContainer href={leavingIBMLink} open />
        </Provider>
      )}

### Browser

Chrome

### Carbon for IBM.com version

10.24.0

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

https://www.ibm.com/it-infrastructure/storage

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://codesandbox.io/s/github/carbon-design-system/carbon-for-ibm-dotcom/tree/master/packages/web-components/examples/codesandbox/components-react/leaving-ibm?file=/src/index.js

### Steps to reproduce the issue (if applicable)

open leaving ibm modal
close it 
open it again

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by kennylam** [comment]
@jeffchew @annawen1 This is something that can be bypassed by setting the state with the [`bx-modal-closed`](https://web-components.carbondesignsystem.com/?path=/docs/components-modal--default#bx-modal-attributes-and-properties). While not ideal, it does unblock this scenario.

A modified example of the included Codesandbox:
https://codesandbox.io/s/cool-darwin-0w9di?file=/src/index.js

This issue has exposed a little bit of feature work we need for this component, namely an embedded link that the `leaving-ibm` component can wrap. This will also make it easier to add multiple external links to a page without needing to target them by selectors.

I'll create a separate issue to track that work, but I think this issue can be closed.
